Assignment/TicTacToe.c

```diff
@@ -0,0 +1,130 @@
+   for (int i = 0; i < 3; i++) {
+      for (int j = 0; j < 3; j++) board[i][j] = EMPTY;
+   }
+}
+
+void PrintBoard (const char board[3][3]) {
+   printf ("\n");
+   for (int i = 0; i < 3; i++) {
+      for (int j = 0; j < 3; j++) {
+         printf (" %c ", board[i][j]);
+         if (j < 2) printf ("|");
+      }
+      printf ("\n");
+      if (i < 2) printf ("---|---|---\n");
+   }
+   printf ("\n");
+}
+
+int CheckWinner (const char board[3][3], char currentPlayer) {
+   for (int i = 0; i < 3; i++) {
+      if ((board[i][0] == board[i][1] && board[i][1] == board[i][2] && board[i][0] == currentPlayer) ||
+          (board[0][i] == board[1][i] && board[1][i] == board[2][i] && board[0][i] == currentPlayer))
+         return 1;
+   }
+   if ((board[0][0] == board[1][1] && board[1][1] == board[2][2] && board[0][0] == currentPlayer) ||
+       (board[0][2] == board[1][1] && board[1][1] == board[2][0] && board[0][2] == currentPlayer))
+      return 1;
+   return 0;
+}
+
+int IsBoardFull (const char board[3][3]) {
+   for (int i = 0; i < 3; i++) {
+      for (int j = 0; j < 3; j++) {
+         if (board[i][j] == EMPTY) return 0;
+      }
+   }
+   return 1;
+}
+
+int MakeMove (char board[3][3], char currentPlayer, int move) {
+   if (move < 1 || move > 9) return 0;
+   int row = (move - 1) / 3, col = (move - 1) % 3;
+   if (board[row][col] == EMPTY) {
+      board[row][col] = currentPlayer;
+      return 1;
+   }
+   return 0;
+}
+
+void SwitchPlayer (char* currentPlayer, char player1Symbol, char player2Symbol) {
+   *currentPlayer = (*currentPlayer == player1Symbol) ? player2Symbol : player1Symbol;
+}
+
+void GetPlayerSymbols (char* player1Symbol, char* player2Symbol) {
+   char symbol;
+   printf ("Player 1, choose your symbol (X/O): ");
+   while (1) {
+      scanf (" %c", &symbol);
+      if (symbol == 'X' || symbol == 'x') {
+         *player1Symbol = 'X';
+         *player2Symbol = 'O';
+         break;
+      }
+      else if (symbol == 'O' || symbol == 'o') {
+         *player1Symbol = 'O';
+         *player2Symbol = 'X';
+         break;
+      }
+      else printf ("Invalid choice! Please choose 'X' or 'O': ");
+   }
+   printf ("Player 2, your symbol is '%c'.\n", *player2Symbol);
+}
+
+int main () {
+   char board[3][3], currentPlayer, player1Symbol, player2Symbol;
+   int gameOver = 0, move = 0;
+   GetPlayerSymbols (&player1Symbol, &player2Symbol);
+   currentPlayer = player1Symbol;
+   InitializeBoard (board);
+   while (!gameOver) {
+      PrintBoard (board);
+      printf ("Player %c, enter a cell (1-9): ", currentPlayer);
+      while (1) {
+         if (scanf ("%d", &move) != 1) {
+            while (getchar () != '\n');
+            printf ("Invalid input! Please enter a number between 1 and 9: ");
+            continue;
+         }
+         if (move < 1 || move > 9) printf ("\nInvalid move! Please enter a number between 1 and 9: ");
+         else {
+            if (MakeMove (board, currentPlayer, move)) break;

```

Comment:
You are already checking if move < 1 || move > 9 here, why check again in MakeMove () ?